### ISSUE TITLE
[BUG] Z-Homing does not put probe at center of build plate

### ISSUE THREAD

**Issue by ScruffR** [MarlinFirmware/Marlin#27491]
### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

I am seeing a very similar problem as described in this closed (but unresolved) bug report from 2020
https://github.com/MarlinFirmware/Marlin/issues/17424

Although I'm pretty sure my settings are correct, when initiating a Z homing cycle, the toolhead ends up at a position offset by -10/-10 of the expected position.

### Expected behavior

When initiating the Z homing cycle, I'd expect the toolhead to move to the position so that the BLTouch probe hits the exact center of the build plate.

### Actual behavior

The toolhead position is off by -10/-10 although I have set these values
```
#define EXTRUDERS 2

#define HOTEND_OFFSET_X { 0.0, -18.00 } // (mm) relative X-offset for each nozzle
#define HOTEND_OFFSET_Y { 0.0, 0.00 }  // (mm) relative Y-offset for each nozzle
#define HOTEND_OFFSET_Z { 0.0, 0.00 }  // (mm) relative Z-offset for each nozzle

#define BLTOUCH

#define NOZZLE_TO_PROBE_OFFSET { -5, -35, 0 }
#define PROBING_MARGIN 6

#define X_BED_SIZE 300
#define Y_BED_SIZE 300

#define X_MIN_POS 0
#define Y_MIN_POS 0
#define Z_MIN_POS 0
#define X_MAX_POS X_BED_SIZE + 20  // allow left nozzle beyond left margin and right nozzle beyond right margin
#define Y_MAX_POS Y_BED_SIZE

#define AUTO_BED_LEVELING_BILINEAR
#define LCD_BED_LEVELING
#define LCD_BED_TRAMMING
  // max possible probe position (relative to E0 home 0/0): LF 6/34 RB 310/277
  #define BED_TRAMMING_INSET_LFRB { 6, 34, X_MAX_POS - 310, Y_MAX_POS - 277 } // (mm) Left, Front, Right, Back insets

#define Z_SAFE_HOMING

#if ENABLED(Z_SAFE_HOMING)
  // Scruffy-CR10 
  // Toolhead: E1 -18/0, E0 0/0, Probe -5/-35
  // Probe position relative to home: LF 6/34 RB 310/277 
  #define Z_SAFE_HOMING_X_POINT 155 // X_CENTER  // (mm) X point for Z homing
  #define Z_SAFE_HOMING_Y_POINT 170 // Y_CENTER  // (mm) Y point for Z homing
#endif
```

### Version of Marlin Firmware

2.1.2.4

### Printer model

modded Crealty CR-10 

### Electronics

GeeeTech GT2560 A+ with DRV8825 drivers

### LCD/Controller

REPRAPWORLD_GRAPHICAL_LCD

### Other add-ons

E3D Chimera dual nozzle, dual extruder, BLTouch

### Bed Leveling

ABL Bilinear mesh

### Your Slicer

Cura

### Host Software

OctoPrint

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

**Comment by ScruffR** [comment]
[Configuration.zip](https://github.com/user-attachments/files/17507886/Configuration.zip)

**Comment by ScruffR** [comment]
In addition to that, when initiating an Automatic Bed Leveling cycle via the OctoPi BedVisualizer add-in the toolhead homes XYZ "correctly" (apart from the -10/-10 offset above) but then when moving to the LF position of the 3x3 measuring mesh the probing positoin does NOT account for the displacement of the BLTouch but tries to probe at 0/0 which should rather be (6-(-5)) / (6-(-35)) or `(PROBING_MARGIN - NOZZLE_TO_PROBE_OFFSET[X]) / (PROBING_MARGIN - NOZZLE_TO_PROBE_OFFSET[Y]).

Since this first point cannot be probed successfully, I can only guess that other "extreme" points of the 3x3 matrix would fail too.

**Comment by ThomasToka** [comment]
Please show the output of M503. And of M114 after "G28" and after "G28 XY".

**Comment by ScruffR** [comment]
That M503 was the crucial hint! 👍
For some reason the EEPROM had 10/10/0 stored and hence overrode/ignored the settings I changed in code
```
Recv: echo:; Z-Probe Offset:
Recv: echo:  M851 X10.00 Y10.00 Z0.00 ; (mm)
```
Since I ported my previous Configuration.h from an rather dated 2.0.x.x I had to go through several iterations till the firmware finally built successfully. By that time I was too eager to flash and forgot to double check all the settings, obviously resulting in an EEPROM re-initialisation with the unaltered/faulty settings.

After resetting the EEPROM with the "final" build running, the correct values show up
```
Recv: echo:; Z-Probe Offset:
Recv: echo:  M851 X-4.99 Y-34.99 Z-0.35 ; (mm) 
```

Thanks for the hint 👍 

Not a bug after all, but user error 😊
